distributed/scheduler.py

```diff
@@ -2130,7 +2130,13 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             worker_pool = self._idle or self._workers
             worker_pool_dv = cast(dict, worker_pool)
             n_workers: Py_ssize_t = len(worker_pool_dv)
-            if n_workers < 20:  # smart but linear in small case
+            # if all occupancies in worker pool of size less than 20
+            # sum to under 0.1 of 1ms; go to the else branch (a round
+            # robin) because the cluster is considered quiet.
+            if (
+                n_workers < 20
+                and sum(w.occupancy for w in worker_pool.values()) > 1.0e-04
```

Comment:
So this is inside of a very tight loop in the scheduler.  We need to be very performance conscious here.  I recommend instead that we call the `min` call below first, and then `if ws.occupancy == 0` we enter some special case.  There are a few things to consider here.  Ideally we want to do a round robin among all of the workers in the pool with zero occupancy.  We could do a full scan through all workers.  We could also zoom to the `n`th task, and then iterate from there to find the next worker with zero occupancy.  This might be constant time in the common case while still robustly choosing zero occupancy in the less common case where things are mixed.

Reply:
The patch as it stands breaks cythonization anyway :) thanks for the input & suggestions